api/src/main/java/org/eclipse/microprofile/reactive/messaging/spi/OutgoingConnectorFactory.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) 2018-2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.eclipse.microprofile.reactive.messaging.spi;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.eclipse.microprofile.reactive.messaging.MessagingProvider;
+import org.eclipse.microprofile.reactive.streams.operators.SubscriberBuilder;
+
+import java.util.NoSuchElementException;
+
+/**
+ * SPI used to implement a connector managing a sink of messages for a specific <em>transport</em>. For example, to
+ * handle dispatch messages to Kafka, the reactive messaging extension would implement a {@code bean} implementing this
+ * interface. This bean is called for every {@code stream} that needs to be created for this specific <em>transport</em>
+ * (so Kafka in this example). These streams are connected to methods annotated with
+ * {@link org.eclipse.microprofile.reactive.messaging.Outgoing}.
+ * <p>
+ * The factory is called to create a {@code subscriber} for each configured <em>transport</em>. The configuration is done
+ * using MicroProfile Config. The following snippet gives an example for a hypothetical Kafka connector:
+ *
+ * <pre>
+ * mp.messaging.outgoing.my-channel.type=i.e.m.reactive.messaging.impl.kafka.Kafka
+ * mp.messaging.outgoing.my-channel.bootstrap-servers=localhost:9092
+ * mp.messaging.outgoing.my-channel.topic=my-topic
+ * ...
+ * </pre>
+ * <p>
+ * This configuration keys are structured as follows: {@code mp.messaging.[incoming|outgoing].channel-name.attribute}.
+ * <p>
+ * The {@code channel-name} segment in the configuration key corresponds to the name of the channel used in the
+ * {@code Outgoing} annotation:
+ *
+ * <pre>
+ * &#64;Outgoing("my-channel")
+ * public CompletionStage&lt;String&gt; produce(String s) {
+ *      // ...
+ * }
+ * </pre>
+ * <p>
+ * The set of attributes depend on the connector and transport layer (For example, bootstrap-servers is Kafka specific).
+ * The {@code type} attribute is mandatory and indicates the fully qualified name of the {@link MessagingProvider}
```

Comment:
If the @Connector annotation is not present and the connector attribute is not set the channel is connected internally to the service. If either is present the channel is a connected channel and wired to an
external message source or sink. If both are the `@connector qualifier and the attribure are present, the attributes value will take precedence. The value of this attribute is used to resolve a class that is used as the `@Connector` qualifier class' parameter value.